include/macros.h

```diff
@@ -238,6 +238,9 @@ extern struct GraphicsContext* __gfxCtx;
 
 #endif /* OOT_DEBUG */
 
+#define gSPMATRIX_SET_NEW(pkt, gfxCtx, file, line) \
```

Comment:
I like that this macro exists, but I feel that there should be a better name than "SET_NEW" to explain what the macro is doing. 

My understanding of things is that you have two matrix systems, the fixed point Mtx which is used by the hardware/what f3dex and similar use, and the floating point MtxF system (sys_matrix) that sits above it which exists to mitigate rounding errors.

On the f3dex side of things, you have the projection matrix "stack" of one element, and the modelview matrix stack. When you load vertex data, the top element in each stack will be multiplied and the vertex data will be transformed in place using the resulting matrix. gSPMatrix modifies the top element of the stack specified by G_MTX_PROJECTION or G_MTX_MODELVIEW, with G_MTX_PUSH will save the previous top element on the stack (does nothing if projection "stack") before doing anything, and G_MTX_MUL concatenating the previous top matrix with the new matrix instead of just overwriting the top matrix.

sys_matrix's primary purpose is to emulate the modelview stack using floating point matrices, while also providing useful matrix operations to apply to the top matrix. `Matrix_NewMtx` (`MATRIX_NEW`) is then used to create a compatible fixed point copy of the top matrix that can be safely accessed from the next graphics task. Not sure why creating a new copy of the top modelview matrix has been reduced to being "NEW", and I think that's where the issue really lies.

So at the very least, I think the macro should be named something to the effect of `gSPMATRIX_MODELVIEW_*`, perhaps `gSPMATRIX_MODELVIEW_NEW_TOP`, though I'd like to see other people's input. With MODELVIEW in the name, I don't think we need the word `SET` to express that we're setting the `G_MTX_NOPUSH | G_MTX_LOAD` flags because the typical mode of operation is to allow the CPU to manage the modelview stack.

Reply:
Your concerns are all applicable to the name of the `MATRIX_NEW` macro: what would you rename it to

Reply:
So if sys_matrix's  `Matrix_NewMtx` should really be named like `Matrix_TopNewMtx`, then perhaps it would lean me towards

*  `MATRIX_NEW` -> `MATRIX_TOP_NEW_MTX`
* `gSPMATRIX_SET_NEW` -> `gSPMATRIX_MODELVIEW_TOP_NEW_MTX`

though I could see NEW being dropped in gSPMATRIX or `NEW_MTX` being combined to `NEWMTX`